extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java

```diff
@@ -1169,4 +1169,9 @@ private List<String> getResourceFiles(Path resourcePath, Path target) {
         }
         return filenames;
     }
+
+    private static boolean isOpenApi_3_1_0_OrGreater(SmallRyeOpenApiConfig config) {
+        final String openApiVersion = config.openApiVersion.orElse(null);
+        return openApiVersion == null || (!openApiVersion.startsWith("2") && !openApiVersion.startsWith("3.0"));
```

Comment:
I'm having a look at this one as I was planning to backport it. When the version is not defined, we consider it's the most recent version?

Reply:
When the version is not defined, it's the latest supported according to my tests. I am not sure if you can backport that as based on this comment https://github.com/quarkusio/quarkus/issues/37199#issuecomment-2302005662 I think it requires https://github.com/quarkusio/quarkus/pull/44055. Better wait for answers from Mike or Phillip.

Reply:
smallrye-open-api will default to the latest depending on the SR version - so `3.0.3` prior to release 4.0, and now `3.1.0` as of 4.0 (#44055). I suspect that may be an issue for backporting, as @michalvavrik already said.